src/shape.rs

```diff
@@ -1014,11 +1017,16 @@ impl ShapeLine {
             vl.spaces += number_of_blanks;
         }
 
+        let first_line_indent = first_line_head_indent.unwrap_or_default().min(line_width);
+
         // This would keep the maximum number of spans that would fit on a visual line
         // If one span is too large, this variable will hold the range of words inside that span
         // that fits on a line.
         // let mut current_visual_line: Vec<VlRange> = Vec::with_capacity(1);
-        let mut current_visual_line = VisualLine::default();
+        let mut current_visual_line = VisualLine {
+            w: first_line_indent,
```

Comment:
Nit: Could be useful to add a comment around _why_ we initialize this to be `first_line_indent`. I think it can be deduced from the surrounding context, but it's not immediately clear

Reply:
Done!